workers.go

```diff
@@ -313,13 +313,6 @@ func (wm *WorkerManager) triggerShutdownIfRequired(decision *FailedDecision) {
 	}
 }
 
-func (wm *WorkerManager) stopBatch() {
```

Comment:
This function wasn't called anywhere, and it scared me a lot - resetting the `currentBatch` to an empty map on the fly with no locking seems really dangerous.